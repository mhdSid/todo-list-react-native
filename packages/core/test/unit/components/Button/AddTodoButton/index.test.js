import { render } from '@testing-library/react-native'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import Component from '../../../../../src/components/Button/AddTodoButton'
import selectors from '../../../../lib/selector/button/addTodoButton'
import MockThemeProvider from '../../../../lib/mocks/MockThemeProvider'

const mockStore = configureStore([thunk])

const store = mockStore({
  todoList: {
    list: []
  },
  auth: {
    user: {
      isLoggedIn: false
    }
  },
  theme: {
    theme: 'light'
  }
})

describe('AddTodoButton', () => {
  test('root exists', async () => {
    const element = render(MockThemeProvider(Component, store))
    const root = await element.findByTestId(selectors.root)
    expect(root).toBeDefined()
  })
  test('title exists', async () => {
    const element = render(MockThemeProvider(Component, store))
    const title = await element.findByTestId(selectors.title)
    expect(title).toBeDefined()
  })
  test('title text', async () => {
    const element = render(MockThemeProvider(Component, store))
    const { children: text } = await element.findByTestId(selectors.title)
    expect(text).toStrictEqual(['+'])
  })
  test('snapshot', () => {
    const data = render(MockThemeProvider(Component, store)).toJSON()
    expect(data).toMatchSnapshot()
  })
})
