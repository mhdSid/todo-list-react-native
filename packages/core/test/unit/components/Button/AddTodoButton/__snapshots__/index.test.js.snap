// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodoButton snapshot 1`] = `
<View
  accessibilityValue={
    Object {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 75,
        "bottom": 30,
        "height": 75,
        "justifyContent": "center",
        "position": "absolute",
        "right": 20,
        "textAlign": "center",
        "width": 75,
        "zIndex": 100,
      },
      Object {
        "backgroundColor": "black",
      },
    ]
  }
  testID="addTodoButton.root"
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 50,
          "marginBottom": 5,
        },
        Object {
          "color": "white",
        },
      ]
    }
    testID="addTodoButton.text"
  >
    +
  </Text>
</View>
`;
